sys/drv/atkbdc.c

```diff
@@ -170,9 +170,9 @@ static int atkbdc_attach(device_t *dev) {
     dev, RT_IOPORTS, 0, IO_KBD, IO_KBD + IO_KBDSIZE - 1, IO_KBDSIZE, RF_ACTIVE);
   assert(atkbdc->regs != NULL);
 
-  atkbdc->intr_handler =
-    INTR_HANDLER_INIT(atkbdc_intr, NULL, atkbdc, "AT keyboard controller", 0);
-  bus_intr_setup(dev, 1, &atkbdc->intr_handler);
+  /* intr_handler name: "AT keyboard controller" */
+  atkbdc->irq_res = bus_alloc_irq(dev, 0, 1 /* magic */, RF_ACTIVE);
```

Comment:
Please don't do magic.
Repos tend to go crazy if there's too much mana power around.

Reply:
We could remove magic with right tools that we don't have at hand, unfortunately. When we implement _resource list_ or similar abstraction the problem will magically go away.

Please refer to http://bxr.su/NetBSD/sys/dev/isa/isavar.h#169, but this is not something we go after.